### PR TITLE
Add a convenience method for editing admin rights and permissions

### PR DESCRIPTION
**Pull request by painor**
Both these methods should work regardless of the entity type(group or channel).
If no user is passed to the edit_permission method the group permission will be the one affected.

### CODE REVIEW

File: telethon/client/chats.py [hunk 1]
```diff
@@ -762,4 +762,229 @@ def action(
...
+                    pin_messages=pin_messages,
+                    add_admins=add_admins
+                )
+            ))
+        elif isinstance(entity, types.InputPeerChat):
+            return await self(functions.messages.EditChatAdminRequest(entity, user, is_admin=is_admin))
+        else:
+            raise ValueError('You must pass either a channel or a supergroup or a normal group')
+
+    async def edit_permission(
```
Comment (Lonami) [inline telethon/client/chats.py]: Should this be renamed as `edit_restriction`? Admin permissions makes more sense, and that's `edit_admin`.

File: telethon/client/chats.py [hunk 2]
```diff
@@ -762,4 +762,229 @@ def action(
...
+                If left as ``None``, the default chat permissions will be updated.
+
+            until_date (`DateLike`, optional):
+                When the user will be unbanned.
+
+                If the due date or duration is longer than 366 days or shorter than
+                30 seconds, the ban will be forever. Defaults to ``0`` (ban forever).
+
+            view_messages (`bool`, optional):
+                Whether the user is able to view messages or not.
```
Comment (Lonami) [inline telethon/client/chats.py]: Are you sure setting this to `False` bans the person? If I remember, it needs to be the opposite for it to have that effect.

Comment (painor) [reply telethon/client/chats.py]: yeah should I inverse them or leave it like telegram does it?

Comment (Lonami) [reply telethon/client/chats.py]: Well, the current documentation is wrong, and if we do it the way Telegram does, it's probably very confusing. So I vote for reversing them.

**Comment by painor** [comment]
oh damn I messed up. I wanted the callback to be another PR. I really need to learn some git.

File: telethon/events/callbackquery.py [hunk 3]
```diff
@@ -26,9 +26,13 @@ class CallbackQuery(EventBuilder):
             can use ``re.compile(b'data_')``.
     """
     def __init__(
-            self, chats=None, *, blacklist_chats=False, func=None, data=None):
+            self, chats=None, *, blacklist_chats=False, func=None, data=None, pattern=None):
```
Comment (Lonami) [inline telethon/events/callbackquery.py]: Oh no, you added this into the same pull request. You should have made a separate branch for this change.

Comment (painor) [reply telethon/events/callbackquery.py]: should be done.

**Comment by Lonami** [comment]
That took a long time to get right.

**Pull request merged**